specification/attestation/data-plane/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json

```diff
@@ -0,0 +1,340 @@
+                    },
+                    {
+                        "$ref": "#/parameters/AuthorizationParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/TeeKind"
+                    },
+                    {
+                        "name": "PolicyJws",
+                        "in": "body",
+                        "description": "JSON Web Signature (See RFC7515) expressing the new policy",
+                        "required": true,
+                        "x-ms-client-flatten": true,
+                        "schema": {
+                            "type": "string"
+                        }
+                    }
+                ],
+                "responses": {
+                  "200": {
+                    "description": "Success - Returns a JWT signed by the metadata signing key that contains the hash of the supplied policy to be set.",
+                    "schema": {
+                        "type": "string"
+                    }
+                  },
+                    "400": {
+                        "description": "Bad request",
+                        "schema": {
+                            "$ref": "#/definitions/CloudError"
+                        }
+                    },
+                    "401": {
+                        "description": "Request is unauthorized",
+                        "schema": {
+                            "type": "string"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed",
+                        "schema": {
+                            "$ref": "#/definitions/CloudError"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "UpdatePolicy": {         
+                      "$ref": "./examples/UpdatePolicy.json"         
+                    }
+                }
+            }
+        },
+        "/operations/policy/current": {
+            "get": {
+                "tags": [
+                         "attestation",
+                         "policy"
+                        ],
+                "summary": "Retrieves the current policy for a given kind of TEE.",
+                "operationId": "Policy_Get",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/AuthorizationParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/TeeKind"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Success",
+                        "schema": {
+                            "$ref": "#/definitions/AttestationPolicy"
```

Comment:
Ideally, the response body for the GET and PUT operations would use the same schema, as would the request body for the PUT request. Is it possible to do so?

Reply:
It can be, however it means that the schema for the return value will be a string and not an object, that may be a challenge for API review.

Reply:
This is a bit tricky. There are three APIs and they each have different semantics. "POST PrepareToUpdate" and "GET" are _almost_ symmetric, but "GET" and "PUT" are not symmetric (the input to "PUT" is the output of "POST PrepareToUpdate". I'm open to changing the URLs involved here, but at it's core, the logical "set" operation requires two separate API calls: PrepareToSet and Set. The "get" logical operation is separate.

Any suggestion about how to resolve this would be appreciated.